### PR TITLE
Plugins: Don't show AT install button once transfer completes

### PR DESCRIPTION
**Pull request by roundhill**
I noticed that the plugin install button was hanging around after a transfer completed. This PR removes the button once the `COMPLETE` status is reached for a transfer.

Video: https://cloudup.com/ccZLLn5dICE

**To test**
* Set up a site for Automated Transfer
* Start the transfer by installing a plugin
* At the end of the transfer, the Success message should display, and the Install button should disappear.

I was trying to think of ways this PR would cause the button to be gone when we don't want it to be. Maybe if a site needs to be transferred again for some reason? Feedback welcome on that.

### CODE REVIEW

**Comment by dmsnell** [comment]
@roundhill does this remove the ability to install _any_ plugin once a single plugin has been installed and an automated transfer has completed?

**Comment by roundhill** [comment]
> does this remove the ability to install any plugin once a single plugin has been installed and an automated transfer has completed?

It doesn't, because we have a special install button just for wpcom plugins before a site is transferred. After the site is transferred, it'll switch over to using the Jetpack plugin install button. Check out: https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/plugins/plugin-meta/index.jsx#L195-L210

File: client/my-sites/plugins-wpcom/plugin-install-button.jsx [hunk 1]
```diff
@@ -24,9 +26,15 @@ export const WpcomPluginInstallButton = props => {
 		siteId,
 		eligibilityData,
 		navigateTo,
-		initiateTransfer
+		initiateTransfer,
+		transferState
 	} = props;
 
+	const { COMPLETE } = transferStates;
+	if ( COMPLETE === transferState ) {
```
Comment (dmsnell) [inline client/my-sites/plugins-wpcom/plugin-install-button.jsx]: this seems a bit redundant, though it's not wrong

if all we're doing though is using it once, why not use compare against `transferStates.COMPLETE`?

**Comment by dmsnell** [comment]
@roundhill it looks like this is only imported in one file - https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/plugins/plugin-meta/index.jsx#L33

Since it's not reused that much how would we feel about performing the state check there instead of from inside the component itself? It's not a clear decision: if we `return null` then we're still rendering a React component and it lives in memory so a condition check in the parent component is cleaner at runtime. On the other hand, if we import it somewhere else and forget the check that leaves the component a bit brittle.

**Comment by roundhill** [comment]
> Since it's not reused that much how would we feel about performing the state check there instead of from inside the component itself?

Sure, I'm cool with that.

**Comment by roundhill** [comment]
@dmsnell all moved over to the parent component in c06bd7b

**Comment by dmsnell** [comment]
hm. I didn't realize that the parent component was calling just another indirection. I hate to say it since you already reworked this @roundhill but I think I liked it better before. if this was just a `{ showAtatButton && <WpComInstallPluginButton … /> }` I would be all for it but here since moving it still results in us rendering a `null` I don't see much value it adds.

**Comment by roundhill** [comment]
Makes sense @dmsnell. We're back to the way it was now with the addition of the constant access you recommended: c06bd7b

File: client/my-sites/plugins-wpcom/plugin-install-button.jsx [hunk 2]
```diff
@@ -24,9 +26,15 @@ export const WpcomPluginInstallButton = props => {
 		siteId,
 		eligibilityData,
 		navigateTo,
-		initiateTransfer
+		initiateTransfer,
+		transferState
 	} = props;
 
+	// Don't show the button if the transfer has completed
```
Comment (dmsnell) [inline client/my-sites/plugins-wpcom/plugin-install-button.jsx]: this is kind of self-explanatory in the code. what if we mentioned _why_ we don't want to show it

```js
// If this button is showing _and_ the transfer is complete then it
// means that this page was visible during the transfer process and
// we no longer want to expose the install button (it's already
// installed by this point). Instead, installs will be handled via the
// normal Jetpack plugins system.
```

**Comment by dmsnell** [comment]
@roundhill one more comment - food for thought - and then I don't see why you shouldn't :shipit:

File: client/my-sites/plugins-wpcom/plugin-install-button.jsx (hunk 2 above)
Comment (roundhill) [inline client/my-sites/plugins-wpcom/plugin-install-button.jsx]: Eh, you're right. I just deleted the comment entirely ;)

**Pull request merged**